### PR TITLE
Fix OOMKilled problem

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. Instead of creating http.Client  in every call to util.DiscoverOAuthEndpoints() - Moved the http.Client creation to phase2_creating and passed the same client as argument. 
2. Called res.Body.Close() when it's empty.

### Issues: Fixed #xxx / Gap #xxx
1. Fixed BZ #1799077

### Testing Instructions:
1.

### CODE REVIEW

File: pkg/system/phase2_creating.go [hunk 1]
```diff
@@ -22,6 +24,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var oauthClient *http.Client
```
Comment (nb-ohad) [inline pkg/system/phase2_creating.go]: Let's save the client on the reconciler, global var seems wrong.

Comment (guymguym) [reply pkg/system/phase2_creating.go]: @nb-ohad @romayalon Wouldn't that cause the same problem though? The reconciler object gets created on every reconcile...

Comment (nb-ohad) [reply pkg/system/phase2_creating.go]: Yes, you are right.

File: pkg/util/util.go [hunk 2]
```diff
@@ -64,8 +68,37 @@ var (
...
+	// SecureHTTPClient is a global secured http client
+	SecureHTTPTransport = newSecureHTTPTransport()
 )
 
+// newSecureHTTPTransport initializes a secured http transport. http.Transport may cause memory leaks when overused
+func newSecureHTTPTransport() *http.Transport {
+	// Load CA cert
+	caCert, err := ioutil.ReadFile(serviceCAPath)
+	if err != nil {
+		return &http.Transport{}
```
Comment (guymguym) [inline pkg/util/util.go]: Do we prefer `return nil` to make it use the default transport?

File: pkg/diagnose/diagnose.go [hunk 3]
```diff
@@ -120,10 +120,7 @@ func (c *Collector) CollectPodLogs(corePodSelector labels.Selector) {
 
 // 	targetAddress := fmt.Sprintf("%s/metrics/counter", mgmtURL.String())
 // 	log.Printf("JENIA THIS IS THE URL %s", targetAddress)
-// 	tr := &http.Transport{
-// 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-// 	}
-// 	client := &http.Client{Transport: tr}
+// 	client := &http.Client{Transport: util.SecureHTTPTransport}
```
Comment (guymguym) [inline pkg/diagnose/diagnose.go]:
```suggestion
// 	client := &http.Client{Transport: util.InsecureHTTPTransport}
```

File: pkg/util/util.go (hunk 2 above)
Comment (romayalon) [reply pkg/util/util.go]: so, returning nil there caused later on a runtime error, even if I checked it for nil and didn't assign anything to the client's transport. returning  &http.Transport{} worked. do you have other suggestion that will be ok ?

Comment (guymguym) [reply pkg/util/util.go]: I'm curious - what is the error when it's nil?

Comment (romayalon) [reply pkg/util/util.go]: Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)

Comment (guymguym) [reply pkg/util/util.go]: of course, but where? what is the stack trace?

**Pull request merged**